Encoder/DefaultEncoder.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Lexik\Bundle\JWTAuthenticationBundle\Encoder;
+
+use InvalidArgumentException;
+use Lexik\Bundle\JWTAuthenticationBundle\Services\JWSProviderInterface;
+use Lexik\Bundle\JWTAuthenticationBundle\Exception\JWTDecodeFailure\ExpiredJWTDecodeFailureException;
+use Lexik\Bundle\JWTAuthenticationBundle\Exception\JWTEncodeFailure\JWTEncodeFailureException;
+use Lexik\Bundle\JWTAuthenticationBundle\Exception\JWTDecodeFailure\JWTDecodeFailureException;
+use Lexik\Bundle\JWTAuthenticationBundle\Exception\JWTDecodeFailure\UnverifiedJWTDecodeFailureException;
+use Lexik\Bundle\JWTAuthenticationBundle\Exception\JWTEncodeFailure\UnsignedJWTEncodeFailureException;
+
+/**
+ * Default Json Web Token encoder/decoder.
+ *
+ * @since 2.0
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class DefaultEncoder extends JWTEncoder
+{
+    /**
+     * @var JWSProviderInterface
+     */
+    protected $jwsProvider;
+
+    /**
+     * @param JWSProviderInterface $jwsProvider
+     */
+    public function __construct(JWSProviderInterface $jwsProvider)
+    {
+        $this->jwsProvider = $jwsProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UnsignedJWTEncodeFailureException
+     */
+    public function encode(array $data)
+    {
+        try {
+            $this->jwsProvider->createSignedToken($data);
+        } catch (InvalidArgumentException $e) {
+            throw new JWTEncodeFailureException('An error occurred while trying to encode the JWT token.', $e);
+        }
+
+        if (JWSProviderInterface::SIGNED !== $this->jwsProvider->getStatus()) {
```

Comment:
The `getStatus()` method isn't part of the `JWSProviderInterface`. And actually, I think we should do this a different way. Having `status` on the `JWSProvider` gives the service "state" - weird things could happen if you were encoding or decoding multiple tokens in a request (I know, edge case). The state should be (somehow) either returned from the `createdSignedToken` and `loadSignature` or an exception should be thrown in this case. But I don't think we should store the status on the service itself.